src/test/ui/deref-suggestion.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/*fn foo(_: String) {}
+
+fn foo2(s: &String) {
+    foo(s);
+}*/
+
+fn foo3(_: u32) {}
+fn foo4(u: &u32) {
+    foo3(u);
+}
+
+fn main() {
+    let s = String::new();
+    let r_s = &s;
+    foo(r_s);
```

Comment:
can you add some comments explaining the purpose of each of these random `foo` functions?

e.g. "// Check that we do not suggest `*r_s` when the argument type is not copy".